src/v/archival/ntp_archiver_service.cc

```diff
@@ -149,12 +173,101 @@ ss::future<> ntp_archiver::upload_loop() {
     }
 }
 
+ss::future<> ntp_archiver::sync_manifest_loop() {
+    ss::lowres_clock::duration backoff = _upload_loop_initial_backoff;
+
+    while (sync_manifest_loop_can_continue()) {
+        auto result = co_await sync_manifest();
+
+        if (result.num_failed != 0) {
+            // The logic in class `remote` already does retries: if we get here,
+            // it means the upload failed after several retries, indicating
+            // something non-transient may be wrong.  Hence error severity.
+            vlog(
+              _rtclog.error,
+              "Failed to upload {} segments out of {}",
+              result.num_failed,
+              result.num_succeded + result.num_failed);
+        } else if (result.num_succeded != 0) {
+            vlog(
+              _rtclog.debug,
+              "Successfuly uploaded {} segments",
+              result.num_succeded);
+        }
+
+        if (!upload_loop_can_continue()) {
```

Comment:
Should this be `if (!sync_manifest_loop_can_continue())` or do we actually care about the upload loop here?

Reply:
Fixed